rsocket-test/src/main/java/io/rsocket/test/LeaksTrackingByteBufAllocator.java

```diff
@@ -201,4 +223,60 @@ public int calculateNewCapacity(int minNewCapacity, int maxCapacity) {
 
     return buffer;
   }
+
+  static final Class<?> simpleLeakAwareCompositeByteBufClass;
+  static final Field leakFieldForComposite;
+  static final Class<?> simpleLeakAwareByteBufClass;
+  static final Field leakFieldForNormal;
+  static final Field allLeaksField;
+
+  static {
+    try {
+      {
+        final Class<?> aClass = Class.forName("io.netty.buffer.SimpleLeakAwareCompositeByteBuf");
+        final Field leakField = aClass.getDeclaredField("leak");
+
+        leakField.setAccessible(true);
+
+        simpleLeakAwareCompositeByteBufClass = aClass;
+        leakFieldForComposite = leakField;
+      }
+
+      {
+        final Class<?> aClass = Class.forName("io.netty.buffer.SimpleLeakAwareByteBuf");
+        final Field leakField = aClass.getDeclaredField("leak");
+
+        leakField.setAccessible(true);
+
+        simpleLeakAwareByteBufClass = aClass;
+        leakFieldForNormal = leakField;
+      }
+
+      {
+        final Class<?> aClass =
+            Class.forName("io.netty.util.ResourceLeakDetector$DefaultResourceLeak");
+        final Field field = aClass.getDeclaredField("allLeaks");
+
+        field.setAccessible(true);
+
+        allLeaksField = field;
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  static Set<Object> resolveTrackingInfo(ByteBuf byteBuf) throws Exception {
+    if (ResourceLeakDetector.getLevel().ordinal()
+        >= ResourceLeakDetector.Level.ADVANCED.ordinal()) {
+      if (simpleLeakAwareCompositeByteBufClass.isInstance(byteBuf)) {
+        return (Set<Object>) allLeaksField.get(leakFieldForComposite.get(byteBuf));
+      } else if (simpleLeakAwareByteBufClass.isInstance(byteBuf)) {
+        return (Set<Object>) allLeaksField.get(leakFieldForNormal.get(byteBuf));
+      }
+    }
```

Comment:
Maybe we should ask Netty if they would consider making it easier to obtain a leak report. Or alternatively, probably better yet, to provide support for tracking leaks in tests.

Reply:
I guess I can port such a tool to Netty, so possibly they accept it eventually and we will use it directly from there

Reply:
To be clear I'm not suggesting that it should be a PR to start. For now it would be valuable enough to raise the idea by creating a well defined issue, also perhaps describing own experience.